gzsitl/CMakeLists.txt

```diff
@@ -1,21 +1,17 @@
 cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
 
-set(CUSTOM_COMPILE_FLAGS "-Wall")
-set(LDFLAGS "-z noexecstack -z relro -z now")
-set(CFLAGS "-fstack-protector-all -fPIE -fPIC -O2 -D_FORTIFY_SOURCE=2")
+set(CUSTOM_COMPILE_FLAGS "${CUSTOM_COMPILE_FLAGS} -Wall")
+set(LDFLAGS "${LDFLAGS} -z noexecstack -z relro -z now")
+set(CFLAGS "${CFLAGS} -fstack-protector-all -fPIE -fPIC -O2 -D_FORTIFY_SOURCE=2")
```

Comment:
That's a good point. I'm not an expert on this matter but from what I've read, in order to successfully generate a position independent executable (PIE), the libraries - both shared and static - that are linked to the executable must be position independent too (compiled with -fPIC). Generating PIEs is a security measure to avoid attacks that depend on knowing beforehand the location where program data will be loaded during execution.